File: src/main/java/gg/mineral/practice/commands/stats/ViewInventoryCommand.java
package gg.mineral.practice.commands.stats;

import gg.mineral.practice.commands.PlayerCommand;
import gg.mineral.practice.inventory.menus.InventoryStatsListMenu;

import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.util.messages.impl.ErrorMessages;
import gg.mineral.practice.util.messages.impl.UsageMessages;
import lombok.val;

public class ViewInventoryCommand extends PlayerCommand {

	public ViewInventoryCommand() {
		super("viewinventory");
	}

	@Override
	public void execute(org.bukkit.entity.Player pl, String[] args) {
		val profile = ProfileManager.getOrCreateProfile(pl);

		if (args.length < 1) {
			profile.message(UsageMessages.VIEW_INV);
			return;
		}

		val inventoryStats = ProfileManager.getInventoryStats(args[0]);

		if (inventoryStats == null || inventoryStats.isEmpty()) {
			profile.message(ErrorMessages.PLAYER_INVENTORY_NOT_FOUND);
			return;
		}

		if (inventoryStats.size() == 1) {
			profile.openMenu(inventoryStats.get(0));
			return;
		}

		profile.openMenu(new InventoryStatsListMenu(inventoryStats));
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/stats/ViewInventoryCommand.kt
package gg.mineral.practice.commands.stats

import dev.rollczi.litecommands.annotations.argument.Arg
import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.inventory.menus.InventoryStatsListMenu
import gg.mineral.practice.managers.ProfileManager.getInventoryStats
import gg.mineral.practice.util.messages.impl.ErrorMessages

@Command(name = "viewinventory")
class ViewInventoryCommand {
    @Execute
    fun execute(@Context profile: Profile, @Arg playerName: String) {
        val inventoryStats = getInventoryStats(playerName)

        if (inventoryStats.isNullOrEmpty()) return profile.message(ErrorMessages.PLAYER_INVENTORY_NOT_FOUND)

        val firstEntry = inventoryStats[0]

        profile.openMenu(
            if (inventoryStats.size == 1) firstEntry else InventoryStatsListMenu(
                inventoryStats,
                firstEntry.opponent
            )
        )
    }
}
